File: lang-notes-schema/api/definition/post.js
const requestBody = {
    type: 'object',
    properties: {
        wordGroup: {
            type: 'object',
            properties: {
                id: { type: 'number' },
                name: { type: 'string' }
            },
            required: []
        },
        clusterId: { type: [ 'number', 'null' ] },
        definition: {
            type: 'object',
            properties: {
                pronounciation: { type: 'string' },
                word: { type: 'string' },
                prefixes: { type: 'string' },
                suffixes: { type: 'string' },
                definition: { type: 'string' }
            },
            required: [ 'word', 'definition' ]
        },
        examples: {
            type: 'array',
            items: {

            }
        },
        translations: {
            type: 'array',
            items: {
                
            }
        }
    },
    required: [ 'wordGroup', 'clusterId', 'definition' ]
}

File: server-rs/src/api/definition/post.rs
use serde::{Deserialize};

#[derive(Debug, Clone, Deserialize)]
pub enum WordGroupProp {
    Id(u32),
    Name(String)
}

#[derive(Debug, Clone, Deserialize)]
pub struct Body {
    pub word_group: WordGroupProp,
    pub cluster_id: u32,
    pub pronounciation: Option<String>,
    pub word: String,
    pub prefixes: Option<String>,
    pub suffixes: Option<String>,
    pub definition: String
}
